### PR TITLE
Add login page.

### PR DESCRIPTION
**Pull request by MFJess**
Why:

* User needs to login to use the app and see weekly task.

This change addresses the need by:

* Reads email and password in input buttons and click the continue button to login. Post request creates a session with token.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,5 +1,123 @@
+import Background from "./assets/Background.svg";
+import Logo from "./assets/spacecenter1.svg";
+import { useNavigate } from "react-router-dom";
+
+async function post(url, body) {
```
Comment (davelange) [inline src/App.js:5]: You're creating this function multiple times in separate files. It's better to create a new file for things related to requests, and export the functions from there. Like that you can always import it from a single place in all the places you need it.

File: src/App.js [hunk 2]
```diff
@@ -1,5 +1,123 @@
...
+  });
+  const json = await response.json();
+  return json;
+}
+
 function App() {
-  return <h1 className="text-3xl font-bold underline">Hello world!</h1>;
+  const navigate = useNavigate();
+
+  const handleSubmit = async (event) => {
```
Comment (davelange) [inline src/App.js:20]: I commented this in previous PRs, but the correct way to get data from a form is FormData (https://github.com/Subvisual-Academy/space/pull/11/#discussion_r1259528137). Also, the inputs in the form need `name` attributes

File: src/assets/Background.svg
Comment (davelange) [inline src/assets/Background.svg:1]: This isn't a big deal, but why is this an svg file? SVGs are best for vector graphics, like logos and icons. This is a bitmap image, so png/jpeg/etc are better suited.

File: src/App.js (hunk 1 above)
Comment (MFJess) [reply src/App.js:5]: You're right! Thanks.

File: src/App.js (hunk 2 above)
Comment (MFJess) [reply src/App.js]: Yes, Tiago had changed that, my code was outdated in some files, but I've already solved that. TY

File: src/pages/login.js [hunk 3]
```diff
@@ -1,5 +1,88 @@
...
+    const mail = document.getElementById("email").value;
+    const pass = document.getElementById("password").value;
+
+    if (mail && pass) {
+      await post(`auth/login`, {
+        email: mail,
+        password: pass,
+      }).then((response) => {
+        current_session["user"] = response["user"];
+        current_session["token"] = response["response"];
```
Comment (davelange) [inline src/pages/login.js:24]: Why are you setting these values on `current_session`, if you're only using them to set local storage?  You could  move the `localStorage.setItem` statements into the `then` callback, and remove `current_session`.

File: src/pages/login.js [hunk 4]
```diff
@@ -1,5 +1,88 @@
+import Background from "../assets/universe_background.jpeg";
+import Logo from "../assets/spacecenter1.svg";
+import { useNavigate } from "react-router-dom";
+import { post } from "../Utils";
+
 function Login() {
-  return <h1>Login</h1>;
+  const navigate = useNavigate();
+
+  const current_session = {
```
Comment (davelange) [inline src/pages/login.js:9]: Just declaring an object like this inside a React component is not very useful, as it basically will be recreated on every render (which happens a lot). If you want the value to persist across renders, you need to store it with `useState` or `useMemo`, though in this case, you probably don't need it at all.

File: src/pages/login.js [hunk 5]
```diff
@@ -1,5 +1,88 @@
...
+    user: "",
+    token: "",
+  };
+
+  const handleSubmit = async (event) => {
+    const mail = document.getElementById("email").value;
```
Comment (davelange) [inline src/pages/login.js:15]: This works, but should use `FormData` here

File: src/pages/login.js [hunk 6]
```diff
@@ -1,5 +1,81 @@
...
+import { useNavigate } from "react-router-dom";
+import { post } from "../Utils";
+
 function Login() {
-  return <h1>Login</h1>;
+  const navigate = useNavigate();
+
+  const handleSubmit = async (event) => {
+    const data = new FormData(event.target);
+    const mail = data.getElementById("email").value;
```
Comment (davelange) [inline src/pages/login.js:11]: If using FormData, and if the inputs have `name` attributes, you can just use the `data.get("email")` method

Comment (davelange) [reply src/pages/login.js:11]: Otherwise, looks good!

**Comment by MFJess** [comment]
I thought I had everything solved, I forgot to solve an error in the code, I havent pushed anything yet. 
Sorry for requesting review :pray:

File: src/pages/login.js [hunk 7]
```diff
@@ -8,8 +8,8 @@ function Login() {
 
   const handleSubmit = async (event) => {
     const data = new FormData(event.target);
-    const mail = data.getElementById("email").value;
-    const pass = data.getElementById("password").value;
+    const mail = data.get("email").value;
```
Comment (davelange) [inline src/pages/login.js:11]: Does this work? If the input has `name="email"` and `name="password"`, then you don't need the `.value` to access the value

Comment (MFJess) [reply src/pages/login.js:11]: Ups you're right. But yes it worked even with .value :sweat_smile:

**Review by davelange — APPROVED** [review]

**Pull request merged**